packages/_private/react-demo/src/DemoForm.tsx

```diff
@@ -0,0 +1,309 @@
+        ? checked
+        : value;
+
+      setFormData((curr) => ({
+        ...curr,
+        [name]: finalValue,
+      }));
+    };
+
+    formRef.current?.addEventListener('syn-change', listener);
+    return () => {
+      formRef.current?.removeEventListener('syn-change', listener);
+    };
+  }, []);
+
+  return (
+    <form
+      onReset={() => {
+        setFormData(initialFormData);
+      }}
+      onSubmit={e => {
+        e.preventDefault();
+        e.stopPropagation();
+
+        const formElement = e.target as HTMLFormElement;
+        const isValid = formElement.checkValidity();
+
+        const content = isValid
+          ? 'Your data was successfully submitted'
+          : 'Your data could not be submitted! Please provide all required information!';
+
+        // eslint-disable-next-line no-alert
+        alert(content);
+      }}
+      ref={formRef}
+    >
+
+      {/* PersonalInformation */}
+      <DemoFieldset legend="Personal Information">
+
+        <SynRadioGroup
+          id="radiogroup-gender"
+          name="gender"
+          label="Please tell us your gender"
+          required
+          value={formData.gender}
+        >
+          <SynRadio value="f">Female</SynRadio>
+          <SynRadio value="m">Male</SynRadio>
+          <SynRadio value="other">Other</SynRadio>
+        </SynRadioGroup>
+
+        <SynSelect
+          id="select-role"
+          label="Current position"
+          name="role"
+          required
+          value={formData.role}
+        >
+          <SynOptgroup label="Developers">
+            <SynOption value="backend">Backend Developer</SynOption>
+            <SynOption value="frontend">Frontend Developer</SynOption>
+          </SynOptgroup>
+          <SynOptgroup label="Other">
+            <SynOption value="lead">Team Lead</SynOption>
+            <SynOption value="other">Other (please specify in comment section below)</SynOption>
+          </SynOptgroup>
+        </SynSelect>
+
+        <SynInput
+          id="input-text"
+          label="Name"
+          minlength={5}
+          maxlength={20}
+          name="name"
+          placeholder="Please insert a value for the regular text input (between 5 and 20 Characters)"
+          required
+          value={formData.name}
+          type="text"
+        />
+
+        <SynInput
+          id="input-email"
+          label="E-Mail"
+          name="email"
+          placeholder="Please insert your E-mail address"
+          required
+          value={formData.email}
+          type="email"
+        />
+
+        <SynInput
+          id="input-phone"
+          label="Phone"
+          name="phone"
+          placeholder="Please provide your phone number"
+          required
+          value={formData.phone}
+          type="tel"
+        />
+
+        <SynInput
+          id="input-date"
+          label="Date of birth"
+          name="date"
+          placeholder="Please insert your E-mail address"
```

Comment:
Should be removed. Is the message from email and a placeholder is not needed for date anyway. This needs to be fixed in all wrappers